File: CoinToss.java
// CoinToss.java — Java версия

import java.util.Scanner;
import java.util.Random;

public class CoinToss {
    private static final String[] COIN_FRAMES = {
        "\n   _____\n  /     \\\n |  (●)  |\n |  (●)  |\n  \\_____/",
        "\n   _____\n  /     \\\n |  ( )  |\n |  ( )  |\n  \\_____/"
    };

    private int heads = 0;
    private int tails = 0;
    private int total = 0;

    private void clearScreen() {
        System.out.print("\033[H\033[2J");
        System.out.flush();
    }

    private void animateCoin(String result) throws InterruptedException {
        for (int i = 0; i < 6; i++) {
            clearScreen();
            System.out.println("🪙 3D Coin Toss (Java)");
            System.out.println("Бросок...");
            System.out.println(COIN_FRAMES[i % 2]);
            Thread.sleep(80);
        }

        clearScreen();
        System.out.println("🪙 3D Coin Toss (Java)");
        if (result.equals("heads")) {
            System.out.println("\u001B[32m");
            System.out.println("   _____");
            System.out.println("  /     \\");
            System.out.println(" |  (●)  |");
            System.out.println(" |  (●)  |");
            System.out.println("  \\_____/");
            System.out.println("\n✅ Результат: ОРЁЛ!\u001B[0m");
            heads++;
        } else {
            System.out.println("\u001B[31m");
            System.out.println("   _____");
            System.out.println("  /     \\");
            System.out.println(" |  ( )  |");
            System.out.println(" |  ( )  |");
            System.out.println("  \\_____/");
            System.out.println("\n❌ Результат: РЕШКА!\u001B[0m");
            tails++;
        }
        total++;
        System.out.printf("\n📊 Статистика: Орёл: %d, Решка: %d\n", heads, tails);
    }

    private void flip() throws InterruptedException {
        Random rand = new Random();
        String result = rand.nextBoolean() ? "heads" : "tails";
        animateCoin(result);
    }

    public void run() throws InterruptedException {
        Scanner scanner = new Scanner(System.in);
        System.out.println("🪙 3D Coin Toss (Java)");
        System.out.println("Нажмите Enter, чтобы бросить монету (q для выхода)");

        while (true) {
            System.out.print("> ");
            String input = scanner.nextLine().trim().toLowerCase();
            if (input.equals("q")) {
                System.out.println("До свидания!");
                break;
            }
            flip();
        }
        scanner.close();
    }

    public static void main(String[] args) throws InterruptedException {
        new CoinToss().run();
    }
}

File: coin_toss.cs
// coin_toss.cs — C# версия

using System;
using System.Threading;

class CoinToss3D
{
    private static readonly string[] COIN_FRAMES = {
        @"
   _____
  /     \
 |  (●)  |
 |  (●)  |
  \_____/",
        @"
   _____
  /     \
 |  ( )  |
 |  ( )  |
  \_____/"
    };

    private int heads = 0;
    private int tails = 0;
    private int total = 0;

    private void ClearScreen()
    {
        Console.Clear();
    }

    private void AnimateCoin(string result)
    {
        for (int i = 0; i < 6; i++)
        {
            ClearScreen();
            Console.WriteLine("🪙 3D Coin Toss (C#)");
            Console.WriteLine("Бросок...");
            Console.WriteLine(COIN_FRAMES[i % 2]);
            Thread.Sleep(80);
        }

        ClearScreen();
        Console.WriteLine("🪙 3D Coin Toss (C#)");
        if (result == "heads")
        {
            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine("   _____");
            Console.WriteLine("  /     \\");
            Console.WriteLine(" |  (●)  |");
            Console.WriteLine(" |  (●)  |");
            Console.WriteLine("  \\_____/");
            Console.WriteLine("\n✅ Результат: ОРЁЛ!");
            Console.ResetColor();
            heads++;
        }
        else
        {
            Console.ForegroundColor = ConsoleColor.Red;
            Console.WriteLine("   _____");
            Console.WriteLine("  /     \\");
            Console.WriteLine(" |  ( )  |");
            Console.WriteLine(" |  ( )  |");
            Console.WriteLine("  \\_____/");
            Console.WriteLine("\n❌ Результат: РЕШКА!");
            Console.ResetColor();
            tails++;
        }
        total++;
        Console.WriteLine($"\n📊 Статистика: Орёл: {heads}, Решка: {tails}");
    }

    private void Flip()
    {
        Random rand = new Random();
        string result = rand.Next(2) == 0 ? "heads" : "tails";
        AnimateCoin(result);
    }

    public void Run()
    {
        Console.WriteLine("🪙 3D Coin Toss (C#)");
        Console.WriteLine("Нажмите Enter, чтобы бросить монету (q для выхода)");

        while (true)
        {
            Console.Write("> ");
            string input = Console.ReadLine()?.Trim().ToLower();
            if (input == "q")
            {
                Console.WriteLine("До свидания!");
                break;
            }
            Flip();
        }
    }

    public static void Main()
    {
        new CoinToss3D().Run();
    }
}
